kolibri/core/content/models.py

```diff
@@ -140,6 +142,38 @@ def __str__(self):
         return self.tag_name
 
 
+class ContentNodeQueryset(TreeQuerySet):
+
+    def dedupe_by_content_id(self):
+        # remove duplicate content nodes based on content_id
+        if OPTIONS['Database']["DATABASE_ENGINE"] == "sqlite":
+            # filter by ids for the deduplicated content nodes
+            deduped_list = []
+            content_ids = set()
+            for node in self:
+                if node.content_id not in content_ids:
+                    deduped_list.append(node)
+                    content_ids.add(node.content_id)
```

Comment:
Not sure how the performance differs, but we could also calculate the node id list with something like this:
`self.values('content_id').annotate(node_id=Min('id')).values_list('node_id', flat=True)`

This might be better than loading all of the nodes into memory.